x/emissions/keeper/msgserver/msg_server_topics.go

```diff
@@ -5,11 +5,60 @@ import (
 
 	errorsmod "cosmossdk.io/errors"
 	alloraMath "github.com/allora-network/allora-chain/math"
-	"github.com/allora-network/allora-chain/x/emissions/types"
+	emissionstypes "github.com/allora-network/allora-chain/x/emissions/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-func (ms msgServer) CreateNewTopic(ctx context.Context, msg *types.MsgCreateNewTopic) (*types.MsgCreateNewTopicResponse, error) {
-	if err := msg.Validate(); err != nil {
+func validateNewTopic(msg *emissionstypes.MsgCreateNewTopic) error {
```

Comment:
If we are adopting this new pattern and moving the validation here, please make sure to move the validations of the other messages as well.